### PR TITLE
item_grid / worlds/ world_details : cleanup + cancel requests, timers ans listeners

### PR DESCRIPTION
**Pull request by aduermael**
- Removed unnecessary `nil` checks used to early return within supposably removed uikit nodes. (checks should be made upstream, same for cancelling requests, timers and listeners)
- Improved friends modal
- fixed ui_container (no more stack overflows)

Fixes CUB-1170
Fixes CUB-1134 (I think it does fix it)
Fixes CUB-1079

### CODE REVIEW

File: lua/modules/api.lua [hunk 1]
```diff
@@ -56,12 +56,31 @@ end
...
+	local url = mod.kApiAddr .. "/users/" .. config.userID .. "/friends"
+
+	if type(fields) == "table" then
+		for i, field in ipairs(fields) do
+			if i == 1 then
+				url = url .. "?f=" .. field
+			else
+				url = url .. "&f=" .. field
+			end
+		end
```
Comment (XLegland) [inline lua/modules/api.lua:80]: Shouldn't we have a function that does this? Takes a table and returns a string, I can see the same thing at line128 (without the `?`)

File: lua/modules/item_details.lua [hunk 2]
```diff
@@ -181,9 +178,6 @@ itemDetails.createModalContent = function(_, config)
 							description.Color = theme.textColor
 							description.pos.Y = descriptionArea.Height - description.Height - theme.padding
 							local req = api:patchItem(itemDetails.id, { description = text }, function(_, _)
-								if removed then
-									return
-								end
 								-- not handling response yet
 							end)
 							table.insert(requests, req)
```
Comment (XLegland) [inline lua/modules/item_details.lua:183]: This block can be outside the if / else

File: lua/modules/api.lua (hunk 1 above)
Comment (aduermael) [reply lua/modules/api.lua:80]: Yes, we'll factorise it at some point. It's a bit out of scope for the PR.

File: lua/modules/item_details.lua (hunk 2 above)
Comment (aduermael) [reply lua/modules/item_details.lua:183]: I agree, adding a commit

Comment (aduermael) [reply lua/modules/item_details.lua]: Done!

**Review by XLegland — APPROVED** [review]
LGTM!

**Pull request merged**